core/src/test/java/org/apache/brooklyn/core/mgmt/rebind/ManagementPlaneIdTest.java

```diff
@@ -0,0 +1,247 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.core.mgmt.rebind;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotEquals;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.Callable;
+
+import org.apache.brooklyn.api.mgmt.ManagementContext;
+import org.apache.brooklyn.api.mgmt.ha.HighAvailabilityMode;
+import org.apache.brooklyn.api.mgmt.ha.ManagementNodeState;
+import org.apache.brooklyn.core.entity.Entities;
+import org.apache.brooklyn.core.internal.BrooklynProperties;
+import org.apache.brooklyn.core.mgmt.internal.LocalManagementContext;
+import org.apache.brooklyn.core.mgmt.persist.BrooklynMementoPersisterToObjectStore;
+import org.apache.brooklyn.core.mgmt.persist.FileBasedObjectStore;
+import org.apache.brooklyn.core.mgmt.persist.PersistMode;
+import org.apache.brooklyn.core.server.BrooklynServerConfig;
+import org.apache.brooklyn.core.server.BrooklynServerPaths;
+import org.apache.brooklyn.test.Asserts;
+import org.apache.brooklyn.util.os.Os;
+import org.apache.brooklyn.util.text.Strings;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class ManagementPlaneIdTest {
+    private File mementoDir;
+
+    protected ClassLoader classLoader = getClass().getClassLoader();
+
+    private Collection<ManagementContext> managementContextForTermination;
+
+    @BeforeMethod
+    public void setUp() {
+        mementoDir = Os.newTempDir(getClass());
+        managementContextForTermination = new ArrayList<>();
+    }
+
+    @AfterMethod(alwaysRun=true)
+    public void tearDown() throws Exception {
+        if (managementContextForTermination != null) {
+            for (ManagementContext mgmt : managementContextForTermination) {
+                Entities.destroyAll(mgmt);
+            }
+        }
+        if (mementoDir != null) FileBasedObjectStore.deleteCompletely(mementoDir);
+    }
+
+    @Test
+    public void testUninitializedThrows() {
+        ManagementContext mgmt = new LocalManagementContext(BrooklynProperties.Factory.newEmpty());
+        assertFalse(mgmt.getOptionalManagementPlaneId().isPresent(), "expected managementPlaneId to be absent");
+    }
+    
+    @Test
+    public void testPlaneIdPersists() throws Exception {
+        final ManagementContext mgmt = createManagementContext(PersistMode.AUTO, HighAvailabilityMode.DISABLED);
+        checkPlaneIdPersisted(mgmt);
+    }
+
+    @Test(enabled=false) //Need to wait for 1hr to verify the update will not overwrite
```

Comment:
Reversed the test to verify that the local planeId is "fixed" by the persisted state so it's quick now.